### PR TITLE
Generating api-spec with Scriptappy

### PR DESCRIPTION
**Pull request by hrigner**

### CODE REVIEW

File: src/qix.js [hunk 1]
```diff
@@ -5,12 +5,75 @@ import SuspendResume from './suspend-resume';
...
+ * @property {Boolean} [protocol.delta=true] Set to false to disable the use of the
+ * bandwidth-reducing delta protocol.
+ */
+
+/**
+ * Mixin object to extend/augment the QIX Engine API
+ * @interface Mixin
+ * @property {String|Array<String>} types String or array of strings containing the API-types that
+ * will be mixed in.
+ * @param {Object} [extend] Object literal containing the methods that will be extended on the
```
Comment (miralemd) [inline src/qix.js]: I think all of these should be `@property` instead of `@param`

File: src/qix.js [hunk 2]
```diff
@@ -5,12 +5,75 @@ import SuspendResume from './suspend-resume';
...
+/**
+ * The API for generated APIs depends on the QIX Engine schema you pass into your Configuration,
+ * and on what QIX struct the API has.
+ * @interface API
+ * @property {String} id Contains the unique identifier for this API.
+ * @property {String} type Contains the schema class name for this API.
+ * @property {String} genericType Corresponds to the qInfo.qType property on the generic object's
+ * properties object.
+ * @property {Object} session Contains a reference to the session that this API belongs to.
+ * @property {String} handle Contains the handle QIX Engine assigned to the API. Used interally in
```
Comment (miralemd) [inline src/qix.js]: handle is a `number`

File: src/qix.js [hunk 3]
```diff
@@ -5,12 +5,75 @@ import SuspendResume from './suspend-resume';
...
+
+/**
+ * The API for generated APIs depends on the QIX Engine schema you pass into your Configuration,
+ * and on what QIX struct the API has.
+ * @interface API
+ * @property {String} id Contains the unique identifier for this API.
+ * @property {String} type Contains the schema class name for this API.
+ * @property {String} genericType Corresponds to the qInfo.qType property on the generic object's
+ * properties object.
+ * @property {Object} session Contains a reference to the session that this API belongs to.
```
Comment (miralemd) [inline src/qix.js]: `Session` as  type

File: src/qix.js (hunk 1 above)
Comment (hrigner) [reply src/qix.js]: Yes! My bad..

File: src/qix.js [hunk 4]
```diff
@@ -43,8 +106,8 @@ class Qix {
 
   /**
   * Function used to create a QIX session.
-  * @param {Object} config The configuration object for the QIX session.
-  * @returns {Session} Returns a new QIX session.
+  * @param {Configuration} config The configuration object for the QIX session.
+  * @returns {Object} Returns a new QIX session.
```
Comment (miralemd) [inline src/qix.js]: `Session`

File: src/intercept.js [hunk 5]
```diff
@@ -5,11 +5,57 @@ import errorResponse from './interceptors/response/error';
...
+
+/**
+ * @class InterceptorResponse
+ * @implements {Interceptor}
+ */
+
+/**
+ * This method is invoked when a request is about to be sent to QIX Engine.
+ * @function InterceptorRequest#onFulfilled
+ * @param {Object} session The session executing the interceptor.
```
[the remark this answers is not in the document]
Comment (hrigner) [reply src/intercept.js]: I fix

File: docs/api-spec.json [hunk 6]
```diff
@@ -0,0 +1,442 @@
+{
+  "scriptappy": "1.0.0",
```
Comment (peol) [inline docs/api-spec.json]: What does this version mean? Is this the API spec version?

Comment (hrigner) [reply docs/api-spec.json]: It's scriptappy schema version..

**Pull request merged**